### PR TITLE
Patch conformance deviations

### PR DESCRIPTION
**Pull request by heshanpadmasiri**
## Purpose
Patch fallowing deviations:
- Maximum negative const int is (-9223372036854775807 - 1) not (-9223372036854775808) ([lang-issue](https://github.com/ballerina-platform/ballerina-lang/issues/32577))
- Update equality to work with singleton typing
- Label tests affected by #1046

### CODE REVIEW

File: conformance/lang/expressions/binary-bitwise-expression/binary_bitwise_and_expr.balt [hunk 1]
```diff
@@ -42,9 +42,9 @@ function init() {
     io:println(6 & -9); // @output 6
     io:println(-43554 & -6); // @output -43558
 
-    io:println(-9223372036854775808 & -9223372036854775808); // @output -9223372036854775808
-    io:println(-9223372036854775808 & 3); // @output 0
-    io:println(-9223372036854775808 & 9223372036854775807); // @output 0
+    io:println((-9223372036854775807 - 1) & (-9223372036854775807 - 1)); // @output -9223372036854775808
```
Comment (MaryamZi) [inline conformance/lang/expressions/binary-bitwise-expression/binary_bitwise_and_expr.balt:45]: We usually use `int:MIN_VALUE` instead. Would that work for nBallerina?

File: conformance/lang/expressions/unary-expression/unary_minus_expression.balt [hunk 2]
```diff
@@ -507,6 +507,7 @@ function init() {
 
 Test-Case: output
 Description: Test unary negative expression for positive floats with no overflow scenarios.
+Fail-Issue: ballerina-platform/ballerina-spec#1046
```
Comment (MaryamZi) [inline conformance/lang/expressions/unary-expression/unary_minus_expression.balt:510]: Usually we break such tests into 2 or more tests, where only the incorrect case will have an associated fail issue in a separate test. We will send a separate PR for this.

Comment (MaryamZi) [reply conformance/lang/expressions/unary-expression/unary_minus_expression.balt:510]: We will also make the tests removed in `boolean_literal.balt` error tests in a separate PR.

File: conformance/lang/expressions/binary-bitwise-expression/binary_bitwise_and_expr.balt (hunk 1 above)
Comment (manuranga) [reply conformance/lang/expressions/binary-bitwise-expression/binary_bitwise_and_expr.balt:45]: We don't support it yet.

Comment (MaryamZi) [reply conformance/lang/expressions/binary-bitwise-expression/binary_bitwise_and_expr.balt:45]: Ack. We need to add the `additive-expr` label.

Comment (heshanpadmasiri) [reply conformance/lang/expressions/binary-bitwise-expression/binary_bitwise_and_expr.balt:45]: fixed

File: conformance/lang/expressions/unary-expression/unary_minus_expression.balt (hunk 2 above)
Comment (heshanpadmasiri) [reply conformance/lang/expressions/unary-expression/unary_minus_expression.balt]: fixed.

File: conformance/lang/expressions/unary-expression/unary_minus_expression.balt [hunk 3]
```diff
@@ -513,27 +513,44 @@ function init() {
...
 
+Test-Case: output
+Description: Test unary negative expression for large positive floats with no overflow scenarios.
+Fail-Issue: ballerina-platform/ballerina-spec#1046
+Labels: DecimalFloatingPointNumber, float, unary-minus
+
+function init() {
+   float a = 922439289.0;
+
+   io:println(-a); // @output -9.22439289E8
```
Comment (MaryamZi) [inline conformance/lang/expressions/unary-expression/unary_minus_expression.balt:530]: Shall we change the output to the correct output?

File: conformance/lang/expressions/unary-expression/unary_minus_expression.balt [hunk 4]
```diff
@@ -513,27 +513,44 @@ function init() {
...
 
+Test-Case: output
+Description: Test unary negative expression for large negative floats with no overflow scenarios.
+Fail-Issue: ballerina-platform/ballerina-spec#1046
+Labels: DecimalFloatingPointNumber, float, unary-minus
+
+function init() {
+   float a = -922439289.0;
+   io:println(-a); // @output 9.22439289E8
+}
```
Comment (MaryamZi) [inline conformance/lang/expressions/unary-expression/unary_minus_expression.balt:553]:
```suggestion
}

```

File: conformance/lang/expressions/unary-expression/unary_minus_expression.balt [hunk 5]
```diff
@@ -513,27 +513,44 @@ function init() {
...
+Description: Test unary negative expression for large positive floats with no overflow scenarios.
+Fail-Issue: ballerina-platform/ballerina-spec#1046
+Labels: DecimalFloatingPointNumber, float, unary-minus
+
+function init() {
+   float a = 922439289.0;
+
+   io:println(-a); // @output -9.22439289E8
+}
+
```
Comment (MaryamZi) [inline conformance/lang/expressions/unary-expression/unary_minus_expression.balt:532]:
```suggestion
```

File: conformance/lang/expressions/literals/boolean_literal.balt [hunk 6]
```diff
@@ -134,10 +140,16 @@ public function main() {
     io:println(false === false); // @output true
 
     io:println(false !== false); // @output false
+}
 
-    io:println(true === false); // @output false
+Test-Case: error
+Description: Exact-equality check of different boolean literals must fail due to singleton typing.
+Fail-Issue: ballerina-platform/ballerina-lang#35450
+Labels: boolean-literal, equality
```
Comment (MaryamZi) [inline conformance/lang/expressions/literals/boolean_literal.balt:148]:
```suggestion
Labels: boolean-literal, exact-equality
```

File: conformance/lang/expressions/unary-expression/unary_minus_expression.balt (hunk 3 above)
Comment (heshanpadmasiri) [reply conformance/lang/expressions/unary-expression/unary_minus_expression.balt]: fixed

File: conformance/lang/expressions/literals/boolean_literal.balt (hunk 6 above)
Comment (heshanpadmasiri) [reply conformance/lang/expressions/literals/boolean_literal.balt]: fixed

File: conformance/lang/expressions/unary-expression/unary_minus_expression.balt (hunk 4 above)
Comment (heshanpadmasiri) [reply conformance/lang/expressions/unary-expression/unary_minus_expression.balt:552]: fixed

**Review by MaryamZi — APPROVED** [review]
